hloc/extractors/r2d2.py

```diff
@@ -0,0 +1,55 @@
+import sys
+from pathlib import Path
+import torch
+import torchvision.transforms as tvf
+
+from ..utils.base_model import BaseModel
+
+r2d2_path = Path(__file__).parent / "../../third_party/r2d2"
+sys.path.append(str(r2d2_path))
+from extract import load_network, NonMaxSuppression, extract_multiscale
+
+RGB_mean = torch.cuda.FloatTensor([0.485, 0.456, 0.406])
+RGB_std  = torch.cuda.FloatTensor([0.229, 0.224, 0.225])
+norm_RGB = tvf.Normalize(mean=RGB_mean, std=RGB_std)
```

Comment:
1. This creates tensors on GPU even if this module is merely imported.
2. hloc should not break on GPU-only machines.

I suggest to:
- create `mean` and `std` in `_init` as `torch.tensor` (CPU) and register them with `register_buffer`. This way, they will be automatically moved to GPU with the model
- call `tvf.functional.normalize` in `_forward` (`mean` and `std` will already be on the correct device)

Reply:
https://github.com/lxxue/Hierarchical-Localization/blob/01b71ee699832a96387e18dc4c70cb6212874772/hloc/extractors/r2d2.py#L28-L29

I pass the mean and std as list as required in this function(https://pytorch.org/docs/stable/_modules/torchvision/transforms/functional.html#normalize) and let the function itself to move the tensors to the suitable device. Is that ok?